flows/contact.go

```diff
@@ -210,6 +208,27 @@ func (c *Contact) HasURN(urn urns.URN) bool {
 	return false
 }
 
+// URNByScheme returns a map of the highest priority URN for each scheme
+func (c *Contact) URNByScheme() types.XMap {
+	byScheme := make(map[string]types.XValue)
+
+	for _, u := range c.urns {
+		scheme := u.URN().Scheme()
+		if _, seen := byScheme[scheme]; !seen {
+			byScheme[scheme] = u
+		}
+	}
+
+	// and add nils for all other schemes
+	for scheme := range urns.ValidSchemes {
```

Comment:
So despite what I argued for previously... giving this map a value for all valid schemes is the only way to replicate the legacy behavior of `@contact.<scheme>`. In a legacy expression that's just empty when contact doesn't have a URN of that scheme, and it's very likely users are relying on that behavior to test contacts to see what URNs they have. Adding empties avoids having to either 1) wrap in `default(..)` or 2) change map behavior and differentiates between `@urns.<validscheme>` and `@urns.<notascheme>` which seems useful. Thoughts?

Reply:
Ya, I'm fine with just having empty values.